docs/site/Deploying_to_ibm_cloud_kubernetes.md

```diff
@@ -0,0 +1,707 @@
+
+````
+Ctrl + C
+````
+
+
+## Step 7: Create a Dockerfile
+
+Let's get the application running in Docker.
+
+In the main project directory, create a file named ````Dockerfile```` .
+
+Place this inside:
+
+````
+FROM node:10.14-alpine
+
+EXPOSE 8080
+
+COPY package.json package.json
+RUN npm install
+
+COPY . .
+
+
+RUN npm run build
+
+CMD ["npm", "start" ]
+````
+
+Notice the port is ````8080```` . It matches the port we specified in the application.
+
+
+
+## Step 8: Build a Docker image
+
+This will build a docker image and give it a tag name.
+
+````
+docker build -t lb4-simple-web-app .
+````
+
+When it completes, you will see :
+
+````
+Successfully built 7d26df6c1561
+Successfully tagged lb4-simple-web-app:latest
+````
+
+You can find your image by typing:
+
+```
+docker images | grep lb4-simple-web-app
+```
+
+It will display something like this :
+```
+lb4-simple-web-app   latest 7d26df6c1561
+```
+
+
+## Step 9: Run the application in Docker
+
+To run the application in a Docker container, type:
+
+````
+docker run -d -i -t  -p 8080:8080 lb4-simple-web-app
+````
+
+This starts the container in detached mode (-d), connects your machine's 8080 port to the 8080 port of the container (-p), and permits you to open a bash shell into the container later on if you'd like (-i -t).
+
+
+Afterwards, type:
+
+````
+docker ps
+````
+
+You should see something like this:
+
+````
+
+CONTAINER ID   IMAGE                COMMAND          CREATED             STATUS         PORTS
+a9962339e863   lb4-simple-web-app   "npm start"      8 seconds ago       Up 7 seconds   0.0.0.0:8080->8080/tcp
+
+````
+
+To see the log output of your container, you can type:
+
+````
+docker logs <container id>    e.g. a9962339e863
+````
+
+You should see something like:
+
+````
+> deploy-test@1.0.0 start /
+> node .
+
+Server is running at http://127.0.0.1:8080
+Try http://127.0.0.1:8080/ping
+````
+
+Open your browser and attempt to access all these urls
+
+http://127.0.0.1:8080/
+
+http://127.0.0.1:8080/ping
+
+http://127.0.0.1:8080/explorer
+
+http://127.0.0.1:8080/openapi.json
+
+
+## Step 10: Stop the application running in Docker
+
+Find the container id
+
+````
+docker ps | grep lb4
+````
+
+You should see something like this:
+
+````
+a9962339e863        lb4-simple-web-app      "npm start"
+````
+
+The leftmost value is the container id.
+
+Type:
+
+````
+docker stop <container id>  e.g. a9962339e863
+````
+
+## Step 11: Comment out the 'EXPOSE 8080' in the Dockerfile
+Before deploying this application to Kubernetes on the IBM Cloud, let's first comment out
+the 'EXPOSE 8080' line in the Dockerfile
+
+````
+#EXPOSE 8080
+````
+
+Build the docker image again:
+
+````
+docker build -t lb4-simple-web-app .
+````
+
+We will be uploading this image to the IBM Cloud Container Registry shortly.
+
+
+## Step 12: Log into IBM Cloud using ibmcloud login command
+
+Use ````ibmcloud login```` command to login.
+
+Side note: Since I’m using a federated user id (IBMid), I use
```

Comment:
Please use "note" or "tip" alert here.

See https://loopback.io/doc/en/contrib/alerts.html#types-of-alerts

Reply:
done